Realm/RLMUtil.mm

```diff
@@ -436,7 +383,68 @@ BOOL RLMIsRunningInPlayground() {
     return [[NSBundle mainBundle].bundleIdentifier hasPrefix:@"com.apple.dt.playground."];
 }
 
-id RLMMixedToObjc(realm::Mixed const& mixed) {
+realm::Mixed RLMObjcToMixed(__unsafe_unretained id v,
+                            __unsafe_unretained RLMRealm *realm,
+                            realm::CreatePolicy createPolicy) {
+    if (!v || v == NSNull.null) {
+        return realm::Mixed();
+    }
+
+    RLMPropertyType type;
+    if ([v conformsToProtocol:@protocol(RLMValue)])
+        type = [v rlm_valueType];
+    else
+        REALM_TERMINATE("Unexpected Type");
+
+    switch (type) {
```

Comment:
I think this whole switch is equivalent to the following:
```
RLMAccessorContext c{objBase->_info};
return switch_on_type(static_cast<realm::PropertyType>(type), [&](auto t) {
        return realm::Mixed(c.unbox<std::decay_t<decltype(*t)>>(v, createPolicy));
});
```

Reply:
this would only work with realm objects as `objBase->_info` holds the class info required for RLMAccessorContext

Reply:
Ah yes, we don't actually have objBase outside of the switch. Pushing that inside would give:

```
return switch_on_type(static_cast<realm::PropertyType>(type), util::overloads{[&](Obj*) {
    RLMAccessorContext c{objBase->_info};
    return realm::Mixed(c.unbox<Obj>(v, createPolicy));
}, [&](auto t) {
    RLMAccessorContext c;
    return realm::Mixed(c.unbox<std::decay_t<decltype(*t)>>(v));
});
```

Awkwardly it isn't possible to construct an accessor context inside the else case. In the property wrappers prototype I extracted [RLMStatelessAccessorContext](https://github.com/realm/realm-cocoa/blob/tg/property-wrapper/Realm/RLMAccessor.hpp#L43) which implements box/unbox on primitives specifically for this sort of thing.

Reply:
I changed things around, @tgoyne if you could take a look